### ISSUE TITLE
[albums] Album sort order

### ISSUE THREAD

**Issue by plegall** [mistic100/Piwigo#286]
**Reported by SnotRocket on 10 Nov 2014 04:15**

**Version:** 2.7.1

Reference: http://piwigo.org/forum/viewtopic.php?id=24347

Piwigo does not sort album names - only image names when selecting sort order as a &quot;user&quot;.

Also, meta data from &quot;image in the album&quot; does not transfer/translate to the album itself.  Example, images in the album are dated 1/1/1970, the album date is created as the date added/create in piwigo, not 1/1/1970.

This throws off the ability to sort albums by date.

**Additional information:** The referenced thread contains most of the information relating to the topic.

In short - piwigo does not currently allow for user sorting ALBUMS, only images within the albums.

[Piwigo Bugtracker #3172](http://piwigo.org/bugs/view.php?id=3172)

**Comment by plegall** [comment]
**Comment posted by jc on 1 Feb 2015 15:56**

There is automatic photo sorting but no automatic album sorting. It took&#039;s me hours to get to know. There are some discussion about it:

http://piwigo.org/forum/viewtopic.php?pid=145691
http://piwigo.org/forum/viewtopic.php?id=24347

And other related bug reports, for example:

http://piwigo.org/bugs/view.php?id=2906 (Album sort order when moving or adding new albums from 2013)
http://piwigo.org/bugs/view.php?id=412 (Auto sort of the categories: add a auto sort per dir name from 2006)

Photo sorting is based on piwigo_categories.image_order SQL column. We need piwigo_categories.cat_order and some hooks, which will rearrange categories after some changes inside.

This is IMHO not only feature request but almost major issue for this type of software :-) Is it possible to get it to 2.8?
